pkgs/development/tools/jdt-language-server/default.nix

```diff
@@ -0,0 +1,72 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, openjdk11_headless
+}:
+
+# Run with: "${pkgs.jdt-language-server}/bin/jdt-language-server"
+# Any additional args will be forwarded (e.g. "-Xms1G")
+stdenv.mkDerivation rec {
+  pname = "jdt-language-server";
+  version = "0.62.0";
+  timestamp = "202009291815";
+
+  src = fetchurl {
+    url = "https://download.eclipse.org/jdtls/milestones/${version}/jdt-language-server-${version}-${timestamp}.tar.gz";
+    sha256 = "05p2frkzl0ikhbr3bjiqxm8gyxg1vn95ci591ig4sl9dbw0k4p8v";
+  };
+
+  sourceRoot = ".";
+
+  buildInputs = [
+    openjdk11_headless
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    # Copy jars
+    install -D -t $out/share/java/plugins/ plugins/*.jar
+
+    # Copy config directories for linux and mac
+    mkdir -p $out/share/config_{linux,mac}/
+    install -D config_linux/* $out/share/config_linux/
+    install -D config_mac/* $out/share/config_mac/
+
+    # Get latest version of launcher jar
+    # e.g. org.eclipse.equinox.launcher_1.5.800.v20200727-1323.jar
+    launcher="$(ls $out/share/java/plugins/org.eclipse.equinox.launcher_* | sort -V | tail -n1)"
+
+    # jdt-language-server needs to write to config location, so we create a temp
+    # directory each time and copy the config_linux/ file there
+    runtime_dir=/tmp/jdt-language-server
```

Comment:
What's this used for ? Should it be XDG_CACHE or something similar?

Reply:
Yeah,  so it's not documented that well, and I wasn't sure if this should be the project directory or some cache directory. I found some examples of how it's run in other packages:

* [emacs-lsp/lsp-java](https://github.com/emacs-lsp/lsp-java/blob/f43b00acd86801aff76c9cdc01f9d7f4b83bf14f/lsp-java.el#L686) passes `$user-emacs-directory/workspace` (I think this would be `~/.config/emacs/workspace`, but I'm not an emacs user so not sure)
* [vscode-java](https://github.com/redhat-developer/vscode-java/blob/78fc024267ff4ce2b1a27de1143c0fb6febec098/src/javaServerStarter.ts#L126) seems to use the workspace directory, which I think isn't exactly the project, but closer to that than a global cache directory.

I actually wonder if it makes more sense to omit of the arguments passed to the jar, since various users and applications may want to do this differently.

Reply:
Correction to the above - it might make sense to omit arguments _after_ `-configuration` and allow the user to specify them.

I believe `-configuration` should be specified by packagers, since it's just a platform-specific directory (e.g. `config_linux/`, `config_mac/`).